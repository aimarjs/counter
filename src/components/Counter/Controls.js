import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Button from '../Button';

const Controls = ({
  onAdd, onReset, onRemove, value,
}) => (
  <Wrapper>
    <Button title="Add" clickHandler={onAdd} />
    <Button title="Reset" clickHandler={onReset} disable={value === 0} />
    <Button title="Remove" clickHandler={onRemove} disable={value === 0} />
  </Wrapper>
);

const Wrapper = styled.div`
  margin: 0 auto;
  max-width: 300px;
`;

Controls.propTypes = {
  onAdd: PropTypes.func.isRequired,
  onReset: PropTypes.func.isRequired,
  onRemove: PropTypes.func.isRequired,
  value: PropTypes.number.isRequired,
};

export default Controls;
