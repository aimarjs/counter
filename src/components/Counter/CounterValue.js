import React from 'react';
import styled from 'styled-components';

const CounterValue = props => (
  <Value>{props.value}</Value>
);

const Value = styled.h2`
  font-size: 64px;
`;

export default CounterValue;
