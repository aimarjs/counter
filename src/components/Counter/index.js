import React from 'react';
import styled from 'styled-components';
import Button from '../Button';
import CounterValue from './CounterValue';

const Counter = props => (
  <Wrapper>
    <CounterValue value={props.value} />
    <Controls>
      <Button title="Add" clickHandler={props.onAdd} />
      <Button title="Reset" clickHandler={props.onReset} disable={props.value === 0} />
      <Button title="Remove" clickHandler={props.onRemove} disable={props.value === 0} />
    </Controls>
  </Wrapper>
);

const Wrapper = styled.div`
  width: 100%
`;

const Controls = styled.div`
  margin: 0 auto;
  max-width: 300px;
`;

export default Counter;
