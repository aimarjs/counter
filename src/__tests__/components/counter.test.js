import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Counter from '../../components/counter'

describe('<Counter />', () => {
  describe('render()', () => {

    it('renders the component', () => {

      const wrapper = shallow(<Counter />);
      const component = wrapper.dive();

      expect(toJson(component)).toMatchSnapshot();

    })

  })
})
