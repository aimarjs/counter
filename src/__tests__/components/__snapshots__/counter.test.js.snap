// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> render() renders the component 1`] = `
<div
  className="sc-bwzfXH fSAIos"
>
  <Button
    title="Add"
  />
  <Button
    disable={false}
    title="Reset"
  />
  <Button
    disable={false}
    title="Remove"
  />
</div>
`;
